airbyte-cdk/bulk/toolkits/load-object-storage/src/main/kotlin/io/airbyte/cdk/load/message/object_storage/ObjectStorageBatch.kt

```diff
@@ -0,0 +1,34 @@
+/*
+ * Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.cdk.load.message.object_storage
+
+import io.airbyte.cdk.load.file.object_storage.Part
+import io.airbyte.cdk.load.file.object_storage.RemoteObject
+import io.airbyte.cdk.load.message.Batch
+
+sealed interface ObjectStorageBatch : Batch
+
+// An indexed bytearray containing an uploadable chunk of a file.
+// Returned by the batch accumulator after processing records.
+class LoadablePart(val part: Part) : ObjectStorageBatch {
+    override val groupId = null
+    override val state = Batch.State.LOCAL
+}
+
+// An UploadablePart that has been uploaded to an incomplete object.
+// Returned by processBatch
+data class IncompletePartialUpload(val key: String) : ObjectStorageBatch {
+    override val state: Batch.State = Batch.State.LOCAL
+    override val groupId: String = key
+}
+
+// An UploadablePart that has triggered a completed upload.
+data class LoadedObject<T : RemoteObject<*>>(
```

Comment:
The nomenclature around complete batch (final object unit) and part (upload unit) is getting murky. I feel like we want to distinguish between the two canonically instead of having having them share the same class.

Something like record -> part -> batch?

Reply:
Right now `Batch` is the generic term for a unit of work in the workflow generated from a set of records. It's at the CDK level. `UploadablePart: Batch` -> `IncompleteObject|LoadedObject: Batch` is at the object storage toolkit level.

I *thought* we were trending toward calling everything Batch at the CDK level. You're suggesting that instead we  push it down to the toolkit level? Or just assume that `Part`, etc is a first class part of the CDK?

Reply:
There's 2 different classes of thing here. The thing we want to upload (batch, file, what have you) and the actual unit we are uploading (part) at a time. The upload itself has one to many parts. I'm suggesting making this distinction first class so this is easier to follow.